src/gtk/menu.cpp

```diff
@@ -665,12 +665,17 @@ void wxMenuItem::SetGtkLabel()
     guint accel_key;
     GdkModifierType accel_mods;
     wxGetGtkAccel(this, &accel_key, &accel_mods);
-    if (accel_key)
+    if ( accel_key && gtk_accelerator_valid( accel_key, accel_mods ) )
     {
         gtk_widget_add_accelerator(
             m_menuItem, "activate", GetRootParentMenu(m_parentMenu)->m_accel,
             accel_key, accel_mods, GTK_ACCEL_VISIBLE);
     }
+    else
+    {
+        // Remove the accelerator since it couldn't be made or is invalid
+        m_text = m_text.BeforeFirst( wxT( '\t' ) );
```

Comment:
```suggestion
        m_text = m_text.BeforeFirst( wxS( '\t' ) );
```
`wxS` is optional, but please don't use `wxT` in the new code.